tenant/store.go

```diff
@@ -76,84 +79,64 @@ func (tenantStore *tenantStore) listSegments(tenantId uint64) ([]Segment, error)
 }
 
 func (tenantStore *tenantStore) addTenant(tenant *Tenant) error {
+	log.Println("In tenantStore addTenant().")
+
 	var tenants []Tenant
 	tenantStore.DbStore.Db.Find(&tenants)
 
-	//	myId := tenant.Id
+	// Most UUID generators return , or "" if uuid is
+	// invalid thus len != 32 should be rejected.
+	// Create new UUID in case one is not provided.
+	if len(tenant.UUID) != 32 {
```

Comment:
As far as I know the state of the discussion, namespaces == tenants in k8s. Isolation is switched on/off on a namespace basis. There is no concept or support for mapping multiple namespaces to a single tenant since they are one and the same.